.github/workflows/email-check.yaml

```diff
@@ -20,14 +20,30 @@ jobs:
 
       - name: Extract author email
         id: author
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
```

Comment:
This is redundant if you're already setting the authorization header manually in the `gh` invocation. You should be able to set this to `GH_TOKEN` and drop the custom header or drop the env variable.